test/cli/flow-sensitivity-hint/test.out

```diff
@@ -0,0 +1,219 @@
+    See https://sorbet.org/docs/flow-sensitive#limitations-of-flow-sensitivity
+    To fix, refactor so the `is_a?` call is on a variable and use that variable everywhere,
+    or accept the autocorrect to silence this error.
+  Autocorrect: Use `-a` to autocorrect
+    test/cli/flow-sensitivity-hint/test.rb:17: Replace with `T.must(x.first)`
+    17 |    x.first.even?
+            ^^^^^^^
+
+test/cli/flow-sensitivity-hint/test.rb:23: Method `even?` does not exist on `NilClass` component of `T.nilable(Integer)` https://srb.help/7003
+    23 |  x[0].even? if !x[0].nil?
+               ^^^^^
+  Got `T.nilable(Integer)` originating from:
+    test/cli/flow-sensitivity-hint/test.rb:23:
+    23 |  x[0].even? if !x[0].nil?
+          ^^^^
+  Possible misuse of flow-sensitive typing:
+    test/cli/flow-sensitivity-hint/test.rb:23: The preceding condition calls `nil?` on a method, not a variable
+    23 |  x[0].even? if !x[0].nil?
+                         ^^^^
+    Sorbet only tracks flow-sensitive knowledge on variables, not methods.
+    See https://sorbet.org/docs/flow-sensitive#limitations-of-flow-sensitivity
+    To fix, refactor so the `nil?` call is on a variable and use that variable everywhere,
+    or accept the autocorrect to silence this error.
+  Autocorrect: Use `-a` to autocorrect
+    test/cli/flow-sensitivity-hint/test.rb:23: Replace with `T.must(x[0])`
+    23 |  x[0].even? if !x[0].nil?
+          ^^^^
+
+test/cli/flow-sensitivity-hint/test.rb:31: Method `even?` does not exist on `NilClass` component of `T.nilable(Integer)` https://srb.help/7003
+    31 |  x.first.even?
+                  ^^^^^
+  Got `T.nilable(Integer)` originating from:
+    test/cli/flow-sensitivity-hint/test.rb:31:
+    31 |  x.first.even?
+          ^^^^^^^
+  Possible misuse of flow-sensitive typing:
+    test/cli/flow-sensitivity-hint/test.rb:29: The preceding condition calls `nil?` on a method, not a variable
+    29 |  return false if x.first.nil?
+                          ^^^^^^^
+    Sorbet only tracks flow-sensitive knowledge on variables, not methods.
+    See https://sorbet.org/docs/flow-sensitive#limitations-of-flow-sensitivity
+    To fix, refactor so the `nil?` call is on a variable and use that variable everywhere,
+    or accept the autocorrect to silence this error.
+  Autocorrect: Use `-a` to autocorrect
+    test/cli/flow-sensitivity-hint/test.rb:31: Replace with `T.must(x.first)`
+    31 |  x.first.even?
+          ^^^^^^^
+
+test/cli/flow-sensitivity-hint/test.rb:36: Method `even?` does not exist on `NilClass` component of `T.nilable(Integer)` https://srb.help/7003
+    36 |  first_is_even = !x.first.nil? && x.first.even?
+                                                   ^^^^^
+  Got `T.nilable(Integer)` originating from:
+    test/cli/flow-sensitivity-hint/test.rb:36:
+    36 |  first_is_even = !x.first.nil? && x.first.even?
+                                           ^^^^^^^
```

Comment:
- the intrinsic for `<check-and-and>` fires before the check in environment.cc this PR adds, and it also changes the error class. the `<check-and-and>` intrinsic changes `UnknownMethod` errors to `CallAfterAndAnd` errors, meaning that the environment.cc thing never trips for `&&` calls

- the and-and intrinsic is still useful because in that case, we have a very easy autocorrect to suggest, which is to rewrite the call as `x.first&.even?` instead of `x.first && x.first.even?`. it's not easy to know how to build this autocorrect in the general case

- Because the call to `<check-and-and>` is generated by syntactically looking for Ruby's `&&` construct, and not via any non-Symbol-based hacks, it's much more reliable than the mechanism in this PR, which is why i'm comfortable giving it its own error code. the changes in this PR will do the wrong thing for certain cases, so i think it's more prudent to leave the error message as a section on an addition error, rather than attempting to completely throw away the original `UnknownMethod` error.